File: schneider_kalf/src/form/nodes/expressions/string/StringLiteral.ts
import ExpressionVisitor from "../../visitors/ExpressionVisitor";
import Expression from "../Expression";

export default class StringLiteral implements Expression {
  private value: string;

  constructor(value: string) {
    this.value = value;
  }

  accept(visitor: ExpressionVisitor): any {
    return visitor.visitStringLiteral(this);
  }

  getValue(): string {
    return this.value;
  }
}

File: Ioana_Hisham/src/main/java/ql/ast/expressions/literals/StringLiteral.java
package ql.ast.expressions.literals;

import ql.ast.expressions.Expression;
import ql.visitors.ExpressionVisitor;

public class StringLiteral extends Expression {

    private final String value;

    public StringLiteral(int lineNumber, String value) {
        super(lineNumber);
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public <T> T accept(ExpressionVisitor<T> expressionVisitor) {
        return expressionVisitor.visit(this);
    }
}
